From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: mm: memory-failure: correct HWPOISON_INJECT config
Date: Fri, 10 Mar 2023 21:38:43 +0800

Use IS_ENABLED(CONFIG_HWPOISON_INJECT) to check whether or not to enable
HWPoison injector module.

Link: https://lkml.kernel.org/r/20230310133843.76883-1-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
Cc: Kefeng Wang <wangkefeng.wang@huawei.com>
Cc: Miaohe Lin <linmiaohe@huawei.com>
Cc: Naoya Horiguchi <naoya.horiguchi@nec.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/memory-failure.c~mm-memory-failure-correct-hwpoison_inject-config
+++ a/mm/memory-failure.c
@@ -168,7 +168,7 @@ static bool page_handle_poison(struct pa
 	return true;
 }
 
-#if defined(CONFIG_HWPOISON_INJECT) || defined(CONFIG_HWPOISON_INJECT_MODULE)
+#if IS_ENABLED(CONFIG_HWPOISON_INJECT)
 
 u32 hwpoison_filter_enable = 0;
 u32 hwpoison_filter_dev_major = ~0U;
_
